### DISCUSSION TITLE
Creating table in custom page in the resource and get this error

### DISCUSSION THREAD

**Discussion by talhachughtai123** [filamentphp/filament#9590] [Help]
App\Filament\Resources\PriceListResource\Pages\ProductsWithStockPrices::table(): Argument #1 ($table) must be of type Filament\Tables\Table, Filament\Infolists\Infolist given, called in C:\laragon\www\forxtremeStock\vendor\filament\infolists\src\Concerns\InteractsWithInfolists.php on line 54
Here is the code 
<?php

namespace App\Filament\Resources\PriceListResource\Pages;

use App\Filament\Resources\PriceListResource;
use Filament\Resources\Pages\Page;
use Filament\Tables\Table;

class ProductsWithStockPrices extends Page
{
    protected static string $resource = PriceListResource::class;

    protected static string $view = 'filament.resources.price-list-resource.pages.products-with-stock-prices';
    public function table(Table $table): Table
{
    return $table
        ->columns([
            // ...
        ]);
}

}

**Comment by leandrocfe** [comment]
share the code please

**Comment by RPotkowaVegasystems** [comment]
Did you implement HasTable with useInteractsWithTable?
